content/post/cars-and-can.md

```diff
@@ -0,0 +1,159 @@
++++
+date = "2017-10-21T13:37:00+01:00"
+description = "Use a cars CAN bus to read live data."
+title = "TED2: Cars and CAN"
+series = ["Ted"]
+slug = "cars-and-can"
+author = "Auke Willem Oosterhoff"
++++
+
+In [previous post]({{< relref "post/obd_ii_elm327_and_python.md">}}) of this
+series I demonstrated how to obtain live data of a car by using an OBD-II
+adapter. I was a bit disappointend about the fact that I could only read a
+handfull of parameters. Together with Wisse Hooghiem I went a step further by
+hooking up in the CAN network of my car. In this post I'll demonstrate how to
+interact with a car's CAN bus in Linux with help of a an Arduino Uno.
+
+## CAN bus
+
+Every car contains electronic chips to control parts of the car. The are called
+Electronic Control Units (ECUs). I identified a few ECUs in my car
+from 2006. Modern cars have a lot more chips on board.
+
+To list a few of the ECUs:
+
+* Motor Control Unit (MCU) - responsible for things like fuel injection and
+  engine speed monitoring
+* Door Control Unit (DCU) - handles the (un)locking of the doors and the
+  windows
+* Tranmission Control Unit (TCU) - controlls the operations related to the tranmsission
+
+Most ECUs require input from other ECUs to operate. For example the Airbag
+Control Unit requires information about speed, acceleration and occupant
+status. Therefore in most cars the ECUs are connected using a CAN bus.
+CAN is an acronym for Controller Area Network and it has beend developed by the
+German company Bosch. CAN is a simple and cheap 2-wire bus. Its first use in
+cars dates back to 1990.
+
+## Data frames
+
+CAN has 4 packet types. But in this post we're only interested in 1 type: the
```

Comment:
Why are you interested in 1 packet type? What are packages types and why aren't you interested in them? I think the reader would be interested to know.

Reply:
Good point, I left the word `cheap` out.

Reply:
Good point. I added a few lines.